src/writer.jl

```diff
@@ -73,9 +73,21 @@ end
 function _print(io::IO, pair::Pair, level::Int=0, ignore_level::Bool=false)
     key = if pair[1] === nothing
         "null" # this is what the YAML parser interprets as 'nothing'
+    elseif pair[1] isa AbstractString && (
+        occursin('#', pair[1]) ||
+            any(
+                c -> startswith(pair[1], c),
+                [
+                    '{', '}', '[', ']', '&', '*', '#', '?', '|', '-', '<', '>', '=',
+                    '!', '%', '@', ':', '`', ',', '"', '\'',
+                ],
+            )
```

Comment:
Would this be (marginally) faster as a tuple?

Reply:
Or maybe as `first(pair[1]) in Set(....)`?

Reply:
You're right! Changing it to a tuple avoids an allocation and is about twice as fast. The naive `Set` version I tried has terrible performance :sweat_smile: